### ISSUE TITLE
Needs proper documentation!

### ISSUE THREAD

**Issue by CallumCVM** [ZehMatt/zasm#9]
The example code is useful but barely scratches the surface of any kind of practical application. Documentation definitely needed.

**Comment by ZehMatt** [comment]
I agree, considering this is fairly new I still have to come around doing it. Some code is documented but not all of it, at the moment I would look into the tests, they are good examples also.

**Comment by CallumCVM** [comment]
Thanks for the reply @ZehMatt and great work on your project, btw :)
